src/emailconfiguration.cc

```diff
@@ -167,6 +167,42 @@ generate_subject (fty_proto_t *alert, const std::string& priority, const std::st
     return s_generateEmailSubjectActive (alert, priority, extname);
 }
 
+
+std::string getIpAddr()
+{
+    std::string ipAddr = "From: ";
+    char * ret;
+    struct ifaddrs * ifAddrStruct = NULL;
+    struct ifaddrs * ifa = NULL;
+    void * tmpAddrPtr = NULL;
+
+    getifaddrs(&ifAddrStruct);
+
+    for (ifa = ifAddrStruct; ifa != NULL; ifa = ifa->ifa_next) {
+        if (!ifa->ifa_addr) {
+            continue;
+        }
+        // Check IP4 adress
+        if (ifa->ifa_addr->sa_family == AF_INET) { 
+            tmpAddrPtr=&((struct sockaddr_in *)ifa->ifa_addr)->sin_addr;
+            char addressBuffer[INET_ADDRSTRLEN];
+            inet_ntop(AF_INET, tmpAddrPtr, addressBuffer, INET_ADDRSTRLEN);
+            if (ifa->ifa_name == "eth0")
+            {
+                ret = addressBuffer;
+                break;
+            }
+        }
+        
+    }
+    if (ifAddrStruct!=NULL) freeifaddrs(ifAddrStruct);
+    if (ifa!=NULL) freeifaddrs(ifa);
+    ipAddr += ret;
```

Comment:
Use of `ret` after free : its value (`addressBuffer`) allocated on the stack left scope.